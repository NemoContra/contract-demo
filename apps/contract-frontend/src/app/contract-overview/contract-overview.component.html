<form>
  <mat-form-field appearance="standard">
    <mat-label>Filter</mat-label>
    <input matInput [formControl]="searchTermControl" placeholder="Search for name">
  </mat-form-field>
</form>

<div class="mat-elevation-z8">
  <ng-container *ngIf="isError$ | async as errorCode; else success">
    <div class="error">
      Search failed with HTTP-Error {{errorCode}}
    </div>
  </ng-container>
  <ng-template #success>
    <ng-container *ngIf="loaded$ | async; else isLoading">
      <table mat-table [dataSource]="dataSource" matSort>

        <ng-container matColumnDef="id">
          <th mat-header-cell *matHeaderCellDef mat-sort-header> ID</th>
          <td mat-cell *matCellDef="let row"> {{row.id}} </td>
        </ng-container>

        <ng-container matColumnDef="firstname">
          <th mat-header-cell *matHeaderCellDef mat-sort-header> Firstname</th>
          <td mat-cell *matCellDef="let row"> {{row.person.firstname}}</td>
        </ng-container>

        <ng-container matColumnDef="lastname">
          <th mat-header-cell *matHeaderCellDef mat-sort-header> Lastname</th>
          <td mat-cell *matCellDef="let row"> {{row.person.lastname}} </td>
        </ng-container>

        <ng-container matColumnDef="contractType">
          <th mat-header-cell *matHeaderCellDef mat-sort-header> Contract Type</th>
          <td mat-cell *matCellDef="let row"> {{row.type}} </td>
        </ng-container>

        <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
        <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>

        <tr class="mat-row" *matNoDataRow>
          <td class="mat-cell" colspan="4">No contracts found.</td>
        </tr>
      </table>
    </ng-container>
    <ng-template #isLoading>
      <div class="loading">
        <mat-spinner></mat-spinner>
      </div>
    </ng-template>
  </ng-template>

  <mat-paginator [pageSizeOptions]="[5, 10, 25, 100]" [length]="totalElements$ | async"
                 aria-label="Select page of contracts"></mat-paginator>
</div>
