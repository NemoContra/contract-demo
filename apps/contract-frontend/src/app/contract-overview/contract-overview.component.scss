table {
  width: 100%;
}

.mat-form-field {
  font-size: 14px;
  width: 100%;
}

td, th {
  width: 25%;
}

.loading, .error {
  display: flex;
  justify-content: center;
  padding-top: 20px;
}
