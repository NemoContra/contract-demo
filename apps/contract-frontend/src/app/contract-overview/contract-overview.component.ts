import { AfterViewInit, ChangeDetectionStrategy, Component, OnInit, ViewChild } from '@angular/core';
import { ContractsFacade } from './+state/contracts.facade';
import { Observable } from 'rxjs';
import { Contract } from '@contract-demo/api-interfaces';
import { MatSort } from '@angular/material/sort';
import { MatPaginator } from '@angular/material/paginator';
import { MatTableDataSource } from '@angular/material/table';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'contract-demo-contract-overview',
  templateUrl: './contract-overview.component.html',
  styleUrls: ['./contract-overview.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ContractOverviewComponent implements OnInit, AfterViewInit {
  contracts$: Observable<Contract[] | undefined> = this.contractsFacade.allContracts$;
  totalElements$: Observable<number | undefined> = this.contractsFacade.totalElements$;
  loaded$: Observable<boolean> = this.contractsFacade.loaded$;
  isError$: Observable<number | undefined> = this.contractsFacade.getContractsError$;

  displayedColumns: string[] = ['id', 'firstname', 'lastname', 'contractType'];
  dataSource: MatTableDataSource<Contract> = new MatTableDataSource<Contract>();

  searchTermControl = new FormControl<string>('', { nonNullable: true });

  @ViewChild(MatPaginator) paginator: MatPaginator;
  @ViewChild(MatSort) sort: MatSort;

  constructor(private readonly contractsFacade: ContractsFacade) {
  }

  ngOnInit(): void {
    this.contractsFacade.getContracts();

    this.searchTermControl.valueChanges.subscribe(value => this.applyFilter(value))

    this.contractsFacade.allContracts$.subscribe((contracts) => this.dataSource.data = contracts)
  }

  ngAfterViewInit(): void {
    this.dataSource.paginator = this.paginator;
    this.dataSource.sort = this.sort;
  }

  applyFilter(term: string) {
    this.contractsFacade.getContracts({ term });

    if (this.dataSource.paginator) {
      this.dataSource.paginator.firstPage();
    }
  }

}
