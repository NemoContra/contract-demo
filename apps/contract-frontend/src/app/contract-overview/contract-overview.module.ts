import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { ContractOverviewComponent } from './contract-overview.component';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { ContractsEffects } from './+state/contracts.effects';
import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatFormFieldModule } from '@angular/material/form-field';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatInputModule } from '@angular/material/input';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import {
  CONTRACTS_FEATURE_KEY,
  contractsReducer,
} from './+state/contracts.reducer';

@NgModule({
  declarations: [ContractOverviewComponent],
  imports: [
    FormsModule,
    ReactiveFormsModule,
    CommonModule,
    RouterModule.forChild([{ path: '', component: ContractOverviewComponent }]),
    StoreModule.forFeature(CONTRACTS_FEATURE_KEY, contractsReducer),
    EffectsModule.forFeature([ContractsEffects]),
    MatTableModule,
    MatPaginatorModule,
    MatFormFieldModule,
    MatInputModule,
    MatProgressSpinnerModule,
  ],
})
export class ContractOverviewModule {}
