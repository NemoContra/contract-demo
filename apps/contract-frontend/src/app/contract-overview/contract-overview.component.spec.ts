import { Spectator, createComponentFactory } from '@ngneat/spectator';

import { ContractOverviewComponent } from './contract-overview.component';

describe('ContractOverviewComponent', () => {
  let spectator: Spectator<ContractOverviewComponent>;
  const createComponent = createComponentFactory(ContractOverviewComponent);

  it('should create', () => {
    spectator = createComponent();

    expect(spectator.component).toBeTruthy();
  });
});
