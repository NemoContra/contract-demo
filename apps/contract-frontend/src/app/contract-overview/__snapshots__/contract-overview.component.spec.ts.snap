// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractOverviewComponent should show correct initial headline and table 1`] = `
<contract-demo-contract-overview
  contractsFacade={[Function Object]}
  dataSource={[Function MatTableDataSource]}
  destroy$={[Function Subject]}
  displayedColumns={[Function Array]}
  filterResult$={[Function Subject]}
  isError$={[Function Subject]}
  loaded$={[Function Subject]}
  paginator={[Function MatPaginator]}
  searchTermControl={[Function FormControl2]}
>
  <h1>
    Contracts overview
  </h1><form
    novalidate=""
  >
    <mat-form-field
      appearance="outline"
      class="mat-mdc-form-field mat-mdc-form-field-type-mat-input mat-form-field-no-animations mat-form-field-appearance-outline mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-valid"
    >
      <div
        class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--outlined"
      >
        <div
          class="mat-mdc-form-field-flex"
        >
          <div
            class="mdc-notched-outline mdc-notched-outline--upgraded"
            matformfieldnotchedoutline=""
          >
            <div
              class="mdc-notched-outline__leading"
            />
            <div
              class="mdc-notched-outline__notch"
            >
              <label
                class="mdc-floating-label mat-mdc-floating-label"
                matformfieldfloatinglabel=""
                style="transition-duration: 0s;"
              >
                <mat-label>
                  Filter
                </mat-label>
              </label>
            </div>
            <div
              class="mdc-notched-outline__trailing"
            />
          </div>
          <div
            class="mat-mdc-form-field-infix"
          >
            <input
              aria-invalid="false"
              aria-required="false"
              class="mat-mdc-input-element ng-untouched ng-pristine ng-valid mat-mdc-form-field-input-control mdc-text-field__input cdk-text-field-autofill-monitored"
              matinput=""
              placeholder="Search for name"
            />
          </div>
        </div>
      </div>
      <div
        class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
      >
        <div
          class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          <div
            class="mat-mdc-form-field-hint-spacer"
          />
        </div>
      </div>
    </mat-form-field>
  </form><div
    class="mat-elevation-z8"
  >
    <div
      class="loading"
    >
      <mat-spinner
        aria-valuemax="100"
        aria-valuemin="0"
        class="mat-mdc-progress-spinner mdc-circular-progress mat-primary _mat-animation-noopable mdc-circular-progress--indeterminate"
        mode="indeterminate"
        role="progressbar"
        style="width: 100px; height: 100px;"
        tabindex="-1"
      >
        <div
          aria-hidden="true"
          class="mdc-circular-progress__determinate-container"
        >
          <svg
            class="mdc-circular-progress__determinate-circle-graphic"
            focusable="false"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="mdc-circular-progress__determinate-circle"
              cx="50%"
              cy="50%"
              r="45"
              style="stroke-dasharray: 282.7433388230814px; stroke-width: 10%;"
            />
          </svg>
        </div>
        <div
          aria-hidden="true"
          class="mdc-circular-progress__indeterminate-container"
        >
          <div
            class="mdc-circular-progress__spinner-layer"
          >
            <div
              class="mdc-circular-progress__circle-clipper mdc-circular-progress__circle-left"
            >
              <svg
                class="mdc-circular-progress__indeterminate-circle-graphic"
                focusable="false"
                viewBox="0 0 100 100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="45"
                  style="stroke-dasharray: 282.7433388230814px; stroke-dashoffset: 141.3716694115407px; stroke-width: 10%;"
                />
              </svg>
            </div>
            <div
              class="mdc-circular-progress__gap-patch"
            >
              <svg
                class="mdc-circular-progress__indeterminate-circle-graphic"
                focusable="false"
                viewBox="0 0 100 100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="45"
                  style="stroke-dasharray: 282.7433388230814px; stroke-dashoffset: 141.3716694115407px; stroke-width: 10%;"
                />
              </svg>
            </div>
            <div
              class="mdc-circular-progress__circle-clipper mdc-circular-progress__circle-right"
            >
              <svg
                class="mdc-circular-progress__indeterminate-circle-graphic"
                focusable="false"
                viewBox="0 0 100 100"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="50%"
                  cy="50%"
                  r="45"
                  style="stroke-dasharray: 282.7433388230814px; stroke-dashoffset: 141.3716694115407px; stroke-width: 10%;"
                />
              </svg>
            </div>
          </div>
        </div>
      </mat-spinner>
    </div>
    <mat-paginator
      aria-label="Select page of contracts"
      class="mat-mdc-paginator"
      role="group"
    >
      <div
        class="mat-mdc-paginator-outer-container"
      >
        <div
          class="mat-mdc-paginator-container"
        >
          <div
            class="mat-mdc-paginator-page-size"
          >
            <div
              class="mat-mdc-paginator-page-size-label"
            >
               Items per page: 
            </div>
            <mat-form-field
              class="mat-mdc-form-field mat-mdc-paginator-page-size-select mat-mdc-form-field-type-mat-select mat-form-field-no-animations mat-form-field-appearance-outline mat-primary"
            >
              <div
                class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--outlined mdc-text-field--no-label"
              >
                <div
                  class="mat-mdc-form-field-flex"
                >
                  <div
                    class="mdc-notched-outline mdc-notched-outline--no-label"
                    matformfieldnotchedoutline=""
                  >
                    <div
                      class="mdc-notched-outline__leading"
                    />
                    <div
                      class="mdc-notched-outline__notch"
                    >
                    </div>
                    <div
                      class="mdc-notched-outline__trailing"
                    />
                  </div>
                  <div
                    class="mat-mdc-form-field-infix"
                  >
                    <mat-select
                      aria-autocomplete="none"
                      aria-disabled="false"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-mdc-select mat-mdc-select-empty"
                      hidesingleselectionindicator=""
                      role="combobox"
                      tabindex="0"
                    >
                      <div
                        cdk-overlay-origin=""
                        class="mat-mdc-select-trigger"
                      >
                        <div
                          class="mat-mdc-select-value"
                        >
                          <span
                            class="mat-mdc-select-placeholder mat-mdc-select-min-line"
                          >
                          </span>
                        </div>
                        <div
                          class="mat-mdc-select-arrow-wrapper"
                        >
                          <div
                            class="mat-mdc-select-arrow"
                          >
                            <svg
                              focusable="false"
                              height="24px"
                              viewBox="0 0 24 24"
                              width="24px"
                            >
                              <path
                                d="M7 10l5 5 5-5z"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </mat-select>
                  </div>
                </div>
              </div>
              <div
                class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
              >
                <div
                  class="mat-mdc-form-field-hint-wrapper ng-trigger ng-trigger-transitionMessages"
                  style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                >
                  <div
                    class="mat-mdc-form-field-hint-spacer"
                  />
                </div>
              </div>
            </mat-form-field>
          </div>
          <div
            class="mat-mdc-paginator-range-actions"
          >
            <div
              aria-live="polite"
              class="mat-mdc-paginator-range-label"
            >
               0 of 0 
            </div>
            <button
              aria-label="Previous page"
              class="mat-mdc-tooltip-trigger mat-mdc-paginator-navigation-previous mdc-icon-button mat-mdc-icon-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
              disabled="true"
              mat-icon-button=""
              type="button"
            >
              <span
                class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
              />
              <svg
                class="mat-mdc-paginator-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
              <span
                class="mat-mdc-focus-indicator"
              />
              <span
                class="mat-ripple mat-mdc-button-ripple"
                matripple=""
              />
              <span
                class="mat-mdc-button-touch-target"
              />
            </button>
            <button
              aria-label="Next page"
              class="mat-mdc-tooltip-trigger mat-mdc-paginator-navigation-next mdc-icon-button mat-mdc-icon-button _mat-animation-noopable mat-unthemed mat-mdc-button-base"
              disabled="true"
              mat-icon-button=""
              type="button"
            >
              <span
                class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
              />
              <svg
                class="mat-mdc-paginator-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
              <span
                class="mat-mdc-focus-indicator"
              />
              <span
                class="mat-ripple mat-mdc-button-ripple"
                matripple=""
              />
              <span
                class="mat-mdc-button-touch-target"
              />
            </button>
          </div>
        </div>
      </div>
    </mat-paginator>
  </div>
</contract-demo-contract-overview>
`;
